pkg/generate/app/sourcelookup.go

```diff
@@ -44,45 +44,32 @@ type SourceRepository struct {
 // NewSourceRepository creates a reference to a local or remote source code repository from
 // a URL or path.
 func NewSourceRepository(s string) (*SourceRepository, error) {
-	var location *url.URL
-	switch {
-	case strings.HasPrefix(s, "git@"):
-		base := "git://" + strings.TrimPrefix(s, "git@")
-		url, err := url.Parse(base)
-		if err != nil {
-			return nil, err
-		}
-		location = url
+	uri, err := url.Parse(s)
```

Comment:
To be clear, urls of form git@ are not actually URLs, so we should not assume they are.